### PR TITLE
David - Added speech recognition

### PR DESCRIPTION
**Pull request by da411d**
Added spitch rekognishin.

### Usage
```
<SpeechRecognition resultCallback={ callbackFunction } />
```
Component passes text to **resultCallback** function.

### CODE REVIEW

File: src/components/Dashboard/MainInput/MainInput.jsx [hunk 1]
```diff
@@ -41,9 +42,12 @@ export default class MainInput extends React.Component {
               placeholder="Type task name"
               aria-label="Type task name"
               aria-describedby="basic-addon2"
-              onChange={val => this.updateInputValue(val)}
+              onChange={event => this.setInputValue(event.target.value)}
               value={newTaskVal}
             />
+            <InputGroup.Append>
+              <SpeechRecognition resultCallback={this.setInputValue}/>
```
Comment (mishalisovskiy) [inline src/components/Dashboard/MainInput/MainInput.jsx]: Please read an article on variables naming. I can recommend this one: https://ru.hexlet.io/blog/posts/naming-errors-1

From what I see here, I absolutely cannot understand what `resultCallback` can stand for.

File: src/components/Dashboard/SpeechRecognition/SpeechRecognition.jsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
+import React, { Component } from 'react';
```
Comment (mishalisovskiy) [inline src/components/Dashboard/SpeechRecognition/SpeechRecognition.jsx]: Please rewrite your code. We use spaces for identation, not tabs. 1 tab = 2 spaces. https://github.com/airbnb/javascript#whitespace

I suggest you read the entire style guide through.

File: src/components/Dashboard/SpeechRecognition/SpeechRecognition.jsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+import OverlayTrigger from "react-bootstrap/OverlayTrigger";
+import Tooltip from "react-bootstrap/Tooltip";
+import Button from "react-bootstrap/Button";
+import Spinner from "react-bootstrap/Spinner";
```
Comment (mishalisovskiy) [inline src/components/Dashboard/SpeechRecognition/SpeechRecognition.jsx]: Please import multiple objects in the single import line.

File: src/components/Dashboard/MainInput/MainInput.jsx [hunk 4]
```diff
@@ -46,7 +46,7 @@ export default class MainInput extends React.Component {
               value={newTaskVal}
             />
             <InputGroup.Append>
-              <SpeechRecognition resultCallback={this.setInputValue}/>
+              <SpeechRecognition onResultReady={this.setInputValue}/>
```
Comment (mishalisovskiy) [inline src/components/Dashboard/MainInput/MainInput.jsx]: Still not clear what this function is intended to do. Please choose names unambiguously.

File: src/components/Dashboard/SpeechRecognition/sendAudioToWatson.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        method: 'POST',
+        headers,
+        body: blob,
+      },
+    )
+      .then(res => res.json());
+  } catch (e) {
+    return false;
+  }
+  result = result.results.filter(res => res.final);
```
Comment (opanki) [inline src/components/Dashboard/SpeechRecognition/sendAudioToWatson.js]: please check this part - if there's no property results in result -> user will get error

**Pull request closed without merging**